# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1368514164 14400
#      Tue May 14 02:49:24 2013 -0400
# Node ID 591cb51c18a3e735065d1e7346142cd318587d99
# Parent  c1d5bf2eb3b6448e70b4c26b82f33cc5be7f13ee
always handle CTRL-C as interrupt on windows systems

* QTerminal.h (QTerminal::QTerminal): Don't set shortcuts for copy and
paste.
* octave-qt-link.cc (octave_qt_link::do_pre_input_event,
octave_qt_link::do_post_input_event): Don't enable/disable processed
input on Windows systems.

diff --git a/libgui/qterminal/libqterminal/QTerminal.h b/libgui/qterminal/libqterminal/QTerminal.h
--- a/libgui/qterminal/libqterminal/QTerminal.h
+++ b/libgui/qterminal/libqterminal/QTerminal.h
@@ -105,22 +105,16 @@ protected:
     QAction *copyAction 
       = _contextMenu->addAction (tr ("Copy"),
                                  this, SLOT (copyClipboard ()));
 
     QAction *pasteAction
       = _contextMenu->addAction (tr ("Paste"),
                                  this, SLOT (pasteClipboard ()));
 
-    copyAction->setShortcut (QKeySequence::Copy);
-    pasteAction->setShortcut (QKeySequence::Paste);
-
-    addAction (copyAction);
-    addAction (pasteAction);
-
     connect (this, SIGNAL (customContextMenuRequested (QPoint)),
              this, SLOT (handleCustomContextMenuRequested (QPoint)));
 
     connect (this, SIGNAL (report_status_message (const QString&)),
              xparent, SLOT (report_status_message (const QString&)));
 
     connect (xparent, SIGNAL (settings_changed (const QSettings *)),
              this, SLOT (notice_settings (const QSettings *)));
diff --git a/libgui/src/octave-qt-link.cc b/libgui/src/octave-qt-link.cc
--- a/libgui/src/octave-qt-link.cc
+++ b/libgui/src/octave-qt-link.cc
@@ -34,22 +34,16 @@ along with Octave; see the file COPYING.
 #include "workspace-element.h"
 #include "builtin-defun-decls.h"
 #include "load-path.h"
 #include "oct-env.h"
 #include "utils.h"
 
 #include "octave-qt-link.h"
 
-#if defined (Q_OS_WIN32)
-#define WIN32_LEAN_AND_MEAN
-#define _WIN32_WINNT 0x0500 
-#include <windows.h>
-#endif
-
 octave_qt_link::octave_qt_link (octave_main_thread *mt)
   : octave_link (), main_thread (mt)
 { }
 
 octave_qt_link::~octave_qt_link (void) { }
 
 void
 octave_qt_link::execute_interpreter (void)
@@ -372,35 +366,21 @@ void
 octave_qt_link::do_clear_history (void)
 {
   emit clear_history_signal ();
 }
 
 void
 octave_qt_link::do_pre_input_event (void)
 {
-#if defined (Q_OS_WIN32)
-  // Disable Ctrl-C processing on stdin.
-  HANDLE hStdIn = GetStdHandle (STD_INPUT_HANDLE);
-  DWORD mode;
-  GetConsoleMode (hStdIn, &mode);
-  SetConsoleMode (hStdIn, mode & ~ ENABLE_PROCESSED_INPUT);
-#endif
 }
 
 void
 octave_qt_link::do_post_input_event (void)
 {
-#if defined (Q_OS_WIN32)
-  // Enable Ctrl-C processing on stdin.
-  HANDLE hStdIn = GetStdHandle (STD_INPUT_HANDLE);
-  DWORD mode;
-  GetConsoleMode (hStdIn, &mode);
-  SetConsoleMode (hStdIn, mode | ENABLE_PROCESSED_INPUT);
-#endif
 }
 
 void
 octave_qt_link::do_enter_debugger_event (const std::string& file, int line)
 {
   do_insert_debugger_pointer (file, line);
 
   emit enter_debugger_signal ();
